[Freerdp] update to <3.3.0>

### Package name

freerdp

### New version number

3.3.0

### Other information that may be useful

It fails with:
```
gcc.exe -DFREERDP_EXPORTS -DNDEBUG -DUNICODE -DWIN32_LEAN_AND_MEAN -DWINVER=0x0601 -DWITH_OPENSSL -DWITH_SMARTCARD_EMULATE -DWITH_VERBOSE_WINPR_ASSERT -D_CRT_SECURE_NO_WARNINGS -D_UNICODE -D_WIN32_WINNT=0x0601 -D_WINSOCK_DEPRECATED_NO_WARNINGS -Dwfreerdp_server_EXPORTS -IC:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/FreeRDP-3.3.0/winpr/include -IC:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/build-UCRT64/winpr/include -IC:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/build-UCRT64 -IC:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/build-UCRT64/include -IC:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/FreeRDP-3.3.0/include -IC:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/FreeRDP-3.3.0/server/Windows/. -march=nocona -msahf -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -Wall -Wpedantic -Wno-padded -Wno-cast-align -Wno-declaration-after-statement -Wall -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -fno-omit-frame-pointer -O3 -DNDEBUG -std=gnu11 -MD -MT server/Windows/CMakeFiles/wfreerdp-server.dir/wf_rdpsnd.c.obj -MF server\Windows\CMakeFiles\wfreerdp-server.dir\wf_rdpsnd.c.obj.d -o server/Windows/CMakeFiles/wfreerdp-server.dir/wf_rdpsnd.c.obj -c C:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/FreeRDP-3.3.0/server/Windows/wf_rdpsnd.c
C:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/FreeRDP-3.3.0/server/Windows/wf_rdpsnd.c: In function 'wf_peer_rdpsnd_activated':
C:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/FreeRDP-3.3.0/server/Windows/wf_rdpsnd.c:78:40: error: 'i' undeclared (first use in this function)
   78 |         context->SelectFormat(context, i);
      |                                        ^
C:/Users/topkek/scoop/persist/msys2/home/topkek/git/MINGW-packages/mingw-w64-freerdp/src/FreeRDP-3.3.0/server/Windows/wf_rdpsnd.c:78:40: note: each undeclared identifier is reported only once for each function it appears in
[13/22] Linking C shared library server\shadow\libfreerdp-shadow3.dll
ninja: build stopped: subcommand failed.

```
maybe another patch on the headers?


### Are you willing to submit a PR?

_No response_

--- a/mingw-w64-freerdp/0002-take-size_t-out-of-the-for-loop-declaration.patch
+++ b/mingw-w64-freerdp/0002-take-size_t-out-of-the-for-loop-declaration.patch
@@ -1,0 +1,27 @@
+From c76088a197d4ccffd719d1e5dcb1c1189f99125d Mon Sep 17 00:00:00 2001
+From: Kreijstal <rainb@tfwno.gf>
+Date: Mon, 26 Feb 2024 15:05:19 +0100
+Subject: [PATCH] take size_t out of the for loop declaration
+
+---
+ server/Windows/wf_rdpsnd.c | 3 ++-
+ 1 file changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/server/Windows/wf_rdpsnd.c b/server/Windows/wf_rdpsnd.c
+index b313c35..dfdbff6 100644
+--- a/server/Windows/wf_rdpsnd.c
++++ b/server/Windows/wf_rdpsnd.c
+@@ -48,8 +48,9 @@ static void wf_peer_rdpsnd_activated(RdpsndServerContext* context)
+ 	wfi = wf_info_get_instance();
+ 	wfi->agreed_format = NULL;
+ 	WLog_DBG(TAG, "Client supports the following %d formats:", context->num_client_formats);
++	size_t i=0;
+ 
+-	for (size_t i = 0; i < context->num_client_formats; i++)
++	for (i = 0; i < context->num_client_formats; i++)
+ 	{
+ 		// TODO: improve the way we agree on a format
+ 		for (size_t j = 0; j < context->num_server_formats; j++)
+-- 
+2.43.2
+

--- a/mingw-w64-freerdp/PKGBUILD
+++ b/mingw-w64-freerdp/PKGBUILD
@@ -3,7 +3,7 @@
 _realname=freerdp
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
-pkgver=3.2.0
+pkgver=3.3.0
 pkgrel=1
 pkgdesc="Free implementation of the Remote Desktop Protocol (RDP) (mingw-w64)"
 arch=('any')
@@ -30,9 +30,11 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-docbook-xsl")
 source=("https://github.com/FreeRDP/FreeRDP/archive/${pkgver}/${_realname}-${pkgver}.tar.gz"
         "0001-fix-some-cmake-warnings.patch"
+        "0002-take-size_t-out-of-the-for-loop-declaration.patch"
         "0008-fix-build-clang-x64.patch")
-sha256sums=('03251e25cc3c8dc688bd9276ce7dc6590058122b192e6c3a15574cb93fc83591'
+sha256sums=('1667af42f8e84bd6e1478fe368c683f7bba6c736655483b0e778b25706bac9b3'
             '0d361f82a5e780530e6c9b5e031079b4db6c8a6561cd0a6e5fb8b2d7edb7a4ea'
+            'ea06dd3b52c44779526dfd91f83e28ef5bdb573f19eaccfaac9aa6afe2024dd5'
             '24e22252f766ea0c0520e12fddcbfb26820dd31200894750938237f52b7c6346')
 
 apply_patch_with_msg() {
@@ -48,6 +50,7 @@ prepare() {
 
   apply_patch_with_msg \
     0001-fix-some-cmake-warnings.patch \
+    0002-take-size_t-out-of-the-for-loop-declaration.patch \
     0008-fix-build-clang-x64.patch
 }
 

